ui/app/js/controllers/release_scheduled_changes_controller.js

```diff
@@ -0,0 +1,164 @@
+angular.module("app").controller("ReleaseScheduledChangesController",
+function($scope, $routeParams, $location, $timeout, Rules, Search, $modal, $route, Releases) {
+
+  $scope.loading = true;
+  $scope.failed = false;
+
+  $scope.sc_id = $routeParams.sc_id;
+
+  function loadPage(newPage) {
+    Releases.getScheduledChangeHistory($scope.sc_id, $scope.pageSize, newPage)
+    .success(function(response) {
+      // it's the same release, but this works
+      $scope.scheduled_changes = response.revisions;
+      $scope.scheduled_changes_count = response.count;
+    })
+    .error(function() {
+      console.error(arguments);
+      $scope.failed = true;
+    })
+    .finally(function() {
+      $scope.loading = false;
+    });
+  }
+
+  if ($scope.sc_id) {
+    $scope.$watch("currentPage", function(newPage) {
+      loadPage(newPage);
+    });
+  } else {
+
+  Releases.getScheduledChanges()
+  .success(function(response) {
+    // "when" is a unix timestamp, but it's much easier to work with Date objects,
+    // so we convert it to that before rendering.
+    $scope.scheduled_changes = response.scheduled_changes.map(function(sc) {
+      if (sc.when !== null) {
+        sc.when = new Date(sc.when);
+      }
+      return sc;
+    });
+  })
+  .error(function() {
+    console.error(arguments);
+    $scope.failed = true;
+  })
+  .finally(function() {
+    $scope.loading = false;
+  });
+}
+  $scope.$watch("ordering_str", function(value) {
+    $scope.ordering = value.value.split(",");
+  });
+  if ($scope.sc_id) {
+    $scope.ordering_options = [
+      {
+        text: "Data Version",
+        value: "-data_version"
+      },
+    ];
+  } else {
+  $scope.ordering_options = [
+    {
+      text: "When",
+      value: "when"
+    },
+    {
+      text: "Product, Name",
```

Comment:
Close! These should be separate items though. There's really no reason to have a two-part sort, but sorting on either exclusively is useful.